### PR TITLE
Pagination and filter

### PR DESCRIPTION
**Pull request by GSuma21**

### CODE REVIEW

File: projects/lib-shared-modules/src/lib/components/pagination/pagination.component.ts [hunk 1]
```diff
@@ -1,12 +1,27 @@
...
-  styleUrl: './pagination.component.css'
+  styleUrl: './pagination.component.scss'
 })
 export class PaginationComponent {
 
+  @ViewChild(MatPaginator) paginator!: MatPaginator;
+  @Input() totalCount:any;
+  @Input() setPaginatorToFirstpage:any
+  @Output() paginatorChanged = new EventEmitter();
+  pageSize = 5;
```
Comment (ramkumar-ns-dev) [inline projects/lib-shared-modules/src/lib/components/pagination/pagination.component.ts]: any reason why it's hard coded here ?

File: projects/lib-shared-modules/src/lib/components/pagination/pagination.component.ts [hunk 2]
```diff
@@ -1,12 +1,27 @@
...
+  pageSizeOptions = [5,10,20,100];
```
Comment (ramkumar-ns-dev) [inline projects/lib-shared-modules/src/lib/components/pagination/pagination.component.ts]: for this also

File: projects/self-creation-portal/src/app/app.component.html [hunk 3]
```diff
@@ -5,3 +5,5 @@
 <lib-card [list]="list"></lib-card>
 
 <lib-search (searchEvent)="receiveSearchResults($event)" [maxLength]="150"></lib-search>
+
+<lib-pagination totalCount="6" (paginatorChanged)="onPaginatorChange($event)"></lib-pagination>
```
Comment (ramkumar-ns-dev) [inline projects/self-creation-portal/src/app/app.component.html:9]: did you checked the complete pagination flow with json data to check pagination and events ?

File: projects/lib-shared-modules/src/lib/components/pagination/pagination.component.ts [hunk 4]
```diff
@@ -1,12 +1,27 @@
...
+ 
+  onPageChange(event: any) {
+    let data = {
+      page: event.pageIndex + 1,
```
Comment (ramkumar-ns-dev) [inline projects/lib-shared-modules/src/lib/components/pagination/pagination.component.ts:22]: what about previous page routing ?
is there a conditions provided if the maxPage length is 1 only, how will the condition verify that ?
what if current page is 1 and user check on previous page, is that logic handled ?

**Review by ramkumar-pacewisdom — CHANGES_REQUESTED** [review]

File: projects/self-creation-portal/src/app/app.component.html (hunk 3 above)
Comment (GSuma21) [reply projects/self-creation-portal/src/app/app.component.html:9]: yes i have checked its working

File: projects/lib-shared-modules/src/lib/components/pagination/pagination.component.ts (hunk 4 above)
Comment (GSuma21) [reply projects/lib-shared-modules/src/lib/components/pagination/pagination.component.ts:22]: mat paginator will handle that , no need to add extra functions.

**Review by ramkumar-pacewisdom — APPROVED** [review]
@kiranharidas187 FYI This PR tested with sample paginations.

**Pull request merged**